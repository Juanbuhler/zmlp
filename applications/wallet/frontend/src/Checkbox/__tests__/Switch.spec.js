import TestRenderer, { act } from 'react-test-renderer'

import CheckboxSwitch from '../Switch'

const noop = () => {}

describe('<Checkbox />', () => {
  it('should render properly', () => {
    const component = TestRenderer.create(
      <CheckboxSwitch
        option={{
          value: 'checkbox',
          label: 'Checkbox',
          initialValue: false,
          isDisabled: false,
        }}
        onClick={noop}
      />,
    )

    // Focus
    act(() => {
      component.root
        .findByProps({ type: 'checkbox' })
        .props.onFocus({ target: { className: 'focus-visible' } })
    })

    expect(component.toJSON()).toMatchSnapshot()

    // Check
    act(() => {
      component.root.findByProps({ type: 'checkbox' }).props.onClick()
    })

    expect(component.toJSON()).toMatchSnapshot()

    // Blur
    act(() => {
      component.root.findByProps({ type: 'checkbox' }).props.onBlur()
    })

    expect(component.toJSON()).toMatchSnapshot()
  })
})
