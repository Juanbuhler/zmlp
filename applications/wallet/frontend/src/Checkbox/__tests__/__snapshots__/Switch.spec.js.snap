// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render properly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 4px;
  background-color: #2e2e2e;
  outline: auto 5px -webkit-focus-ring-color;
}

.emotion-0:hover {
  background-color: #2e2e2e;
}

.emotion-0:hover div {
  color: #ffffff;
}

.emotion-0:hover rect {
  stroke: #ffffff;
}

.emotion-0:hover circle {
  fill: #ffffff;
}

.emotion-0 div {
  color: #ffffff;
}

.emotion-0 rect {
  stroke: #ffffff;
}

.emotion-0 circle {
  fill: #ffffff;
}

.emotion-1 {
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.emotion-2 {
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
  fill: #4a4a4a;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  margin-left: 8px;
  text-transform: uppercase;
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
  color: #808080;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: Roboto Condensed;
  margin-bottom: -1px;
}

<label
  className="emotion-0"
>
  <input
    className="hidden"
    defaultChecked={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value="checkbox"
  />
  <svg
    height={16}
    width={30}
  >
    <rect
      className="emotion-1"
      fill="#191919"
      height={8}
      rx={4}
      stroke="#3b3b3b"
      width={30}
      x={0}
      y={4}
    />
    <circle
      className="emotion-2"
      cx={8}
      cy={8}
      r={8}
    />
  </svg>
  <div
    className="emotion-3"
  >
    Checkbox
  </div>
</label>
`;

exports[`<Checkbox /> should render properly 2`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 4px;
  background-color: #2e2e2e;
  outline: auto 5px -webkit-focus-ring-color;
}

.emotion-0:hover {
  background-color: #2e2e2e;
}

.emotion-0:hover div {
  color: #3A5DFF;
}

.emotion-0:hover rect {
  stroke: #3b3b3b;
}

.emotion-0:hover circle {
  fill: #3A5DFF;
}

.emotion-0 div {
  color: #3A5DFF;
}

.emotion-0 rect {
  stroke: #3b3b3b;
}

.emotion-0 circle {
  fill: #3A5DFF;
}

.emotion-1 {
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.emotion-2 {
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
  fill: #3A5DFF;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  margin-left: 8px;
  text-transform: uppercase;
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
  color: #3A5DFF;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: Roboto Condensed;
  margin-bottom: -1px;
}

<label
  className="emotion-0"
>
  <input
    className="hidden"
    defaultChecked={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value="checkbox"
  />
  <svg
    height={16}
    width={30}
  >
    <rect
      className="emotion-1"
      fill="#191919"
      height={8}
      rx={4}
      stroke="#3b3b3b"
      width={30}
      x={0}
      y={4}
    />
    <circle
      className="emotion-2"
      cx={22}
      cy={8}
      r={8}
    />
  </svg>
  <div
    className="emotion-3"
  >
    Checkbox
  </div>
</label>
`;

exports[`<Checkbox /> should render properly 3`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 4px;
}

.emotion-0:hover {
  background-color: #2e2e2e;
}

.emotion-0:hover div {
  color: #3A5DFF;
}

.emotion-0:hover rect {
  stroke: #3b3b3b;
}

.emotion-0:hover circle {
  fill: #3A5DFF;
}

.emotion-1 {
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.emotion-2 {
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
  fill: #3A5DFF;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  margin-left: 8px;
  text-transform: uppercase;
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
  color: #3A5DFF;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: Roboto Condensed;
  margin-bottom: -1px;
}

<label
  className="emotion-0"
>
  <input
    className="hidden"
    defaultChecked={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value="checkbox"
  />
  <svg
    height={16}
    width={30}
  >
    <rect
      className="emotion-1"
      fill="#191919"
      height={8}
      rx={4}
      stroke="#3b3b3b"
      width={30}
      x={0}
      y={4}
    />
    <circle
      className="emotion-2"
      cx={22}
      cy={8}
      r={8}
    />
  </svg>
  <div
    className="emotion-3"
  >
    Checkbox
  </div>
</label>
`;
